core/container.go

```diff
@@ -1715,6 +1731,47 @@ func (container *Container) WithoutExposedPort(port int, protocol NetworkProtoco
 	return container, nil
 }
 
+func (container *Container) updatePortsFromConfig() *Container {
+	if container.Config.ExposedPorts == nil {
+		return container
+	}
+
+	container = container.Clone()
+
+	// convert to map to preserve the descriptions (not in the OCI spec)
+	existing := make(map[string]*ContainerPort, len(container.Ports))
+	for _, p := range container.Ports {
+		p := p
+		ociPort := fmt.Sprintf("%d/%s", p.Port, p.Protocol.Network())
+		existing[ociPort] = &p
+	}
+
+	ports := make([]ContainerPort, 0, len(container.Config.ExposedPorts))
+	for ociPort := range container.Config.ExposedPorts {
+		p, exists := existing[ociPort]
+		if !exists {
+			// ignore errors when parsing from OCI
+			port, proto, ok := strings.Cut(ociPort, "/")
+			if !ok {
+				continue
+			}
+			portNr, err := strconv.Atoi(port)
+			if err != nil {
+				continue
+			}
+			p = &ContainerPort{
+				Port:     portNr,
+				Protocol: NetworkProtocol(strings.ToUpper(proto)),
+			}
+		}
+		ports = append(ports, *p)
+	}
+
+	container.Ports = ports
```

Comment:
Double-checking, I thought we didn't want to touch `container.Ports` during `From` and `Build`, per https://github.com/dagger/dagger/pull/5158#issuecomment-1576536740?

Reply:
😮 Looks like it was re-added from today's [rebase](https://github.com/dagger/dagger/compare/27311b05a379f65c5ca5b49f451afa3f605297c3..3486da204a215f9c705bd34a217e5a7455e64fcb). Thanks for catching that!

Reply:
This commit seems to have been lost: https://github.com/dagger/dagger/commit/20727b6e653b15328c13b9a3d14a752e94f2b586

Reply:
This commit seems to have been lost: https://github.com/dagger/dagger/commit/20727b6e653b15328c13b9a3d14a752e94f2b586

Reply:
This commit seems to have been lost: https://github.com/dagger/dagger/commit/11c1ece23448b5baa4937bc73e1380c75fc38f08

Reply:
These commits seem to have been lost:
- https://github.com/dagger/dagger/commit/9b02b80a7e96ab17386a7306cc1f29eefff86fa5
- https://github.com/dagger/dagger/commit/27311b05a379f65c5ca5b49f451afa3f605297c3

Reply:
@vito they have been recovered.